presto-main/src/main/java/com/facebook/presto/execution/QueryTracker.java

```diff
@@ -208,11 +213,19 @@ private void enforceTimeLimits()
                 continue;
             }
             Duration queryMaxRunTime = getQueryMaxRunTime(query.getSession());
-            Duration queryMaxExecutionTime = getQueryMaxExecutionTime(query.getSession());
+            Optional<QueryLimit<Duration>> queryMaxExecutionTime = getMinimum(
```

Comment:
Not following why this is now optional?

Reply:
`query.getSession()` was not changed, so this will end up having a value in practice - Just we wanted `query::getResourceGroupQueryLimits()` to be optional as the value is set during runtime, making this a way to compare a potentially empty optional to another value